test/dates/ranges.jl

```diff
@@ -482,10 +482,10 @@ function testmonthranges2(n)
 end
 testmonthranges2(100000)
 
-@test length(Dates.Year(1):Dates.Year(10)) == 10
+@test length(Dates.Year(1):Dates.Year(1):Dates.Year(10)) == 10
```

Comment:
In principle I agree that you should define step for unitful ranges. However since ranges involving periods currently require that the endpoints and the step are of the same unit I think this change is mostly just inconvenient. Overall though I'm happy with the change if it leads to less confusion.